### PR TITLE
Use logger for helm output

### PR DESCRIPTION
**Pull request by inteon**
My previous PR was auto-closed: https://github.com/helmfile/helmfile/pull/1386

This PR updates all helm invocations, so the output is written to stderr via the logger only.
Only when templating, we write directly to stdout.

This PR aims to improve the experience when using helmfile as a go library.

### CODE REVIEW

File: test/integration/test-cases/diff-args/output/apply [hunk 1]
```diff
@@ -67,11 +67,3 @@ helmfile-tests, installed-httpbin, Service (v1) has been added:
 +     release: installed
 +   type: LoadBalancer
 
-Release "installed" does not exist. Installing it now.
```
Comment (inteon) [inline test/integration/test-cases/diff-args/output/apply:70]: These lines are now printed to stderr.

File: test/integration/test-cases/skip-diff-output/output/diff-result [hunk 2]
```diff
@@ -1,4 +1,3 @@
-Comparing release=foo, chart=../../../charts/raw
```
Comment (inteon) [inline test/integration/test-cases/skip-diff-output/output/diff-result:1]: This line is now printed to stderr.

File: test/integration/test-cases/skip-diff-output/output/diff-result-live [hunk 3]
```diff
@@ -13,5 +13,4 @@ helmfile-tests, baz-2, ConfigMap (v1) has been added:
 +   namespace: helmfile-tests
 + data:
 +   baz: BAZ
-Comparing release=foo, chart=../../../charts/raw
```
Comment (inteon) [inline test/integration/test-cases/skip-diff-output/output/diff-result-live:16]: This line is now printed to stderr.

**Comment by yxxhero** [comment]
@mumoshu WDYT?

**Review by mumoshu — CHANGES_REQUESTED** [review]
@inteon Thank you very much for your effort!

This looks almost good. Only two things:

- Can we also make the logging output configurable https://github.com/helmfile/helmfile/blob/442e8647a7d20ec71c54b9523c0a614fdc7681bc/cmd/root.go#L63 so that anyone wants logs to flow into stdout as before can configure to do so for some form of backward-compatibility
- Can we have test outputs for the combo of stdout/stderr, for example, named `apply-combined` and `apply-live-combined` and make tests to check those? The rationale is that, by moving some messages from stdout to stderr, we lose some test coverage. I hope we can keep the same degree of test coverage even after this change.

**Comment by inteon** [comment]
> @inteon Thank you very much for your effort!
> 
> This looks almost good. Only two things:
> 
> * Can we also make the logging output configurable https://github.com/helmfile/helmfile/blob/442e8647a7d20ec71c54b9523c0a614fdc7681bc/cmd/root.go#L63
>    so that anyone wants logs to flow into stdout as before can configure to do so for some form of backward-compatibility
> * Can we have test outputs for the combo of stdout/stderr, for example, named `apply-combined` and `apply-live-combined` and make tests to check those? The rationale is that, by moving some messages from stdout to stderr, we lose some test coverage. I hope we can keep the same degree of test coverage even after this change.



> @inteon Thank you very much for your effort!
> 
> This looks almost good. Only two things:
> 
> * Can we also make the logging output configurable https://github.com/helmfile/helmfile/blob/442e8647a7d20ec71c54b9523c0a614fdc7681bc/cmd/root.go#L63
>    so that anyone wants logs to flow into stdout as before can configure to do so for some form of backward-compatibility
> * Can we have test outputs for the combo of stdout/stderr, for example, named `apply-combined` and `apply-live-combined` and make tests to check those? The rationale is that, by moving some messages from stdout to stderr, we lose some test coverage. I hope we can keep the same degree of test coverage even after this change.



> @inteon Thank you very much for your effort!
> 
> This looks almost good. Only two things:
> 
> * Can we also make the logging output configurable https://github.com/helmfile/helmfile/blob/442e8647a7d20ec71c54b9523c0a614fdc7681bc/cmd/root.go#L63
>    so that anyone wants logs to flow into stdout as before can configure to do so for some form of backward-compatibility
> * Can we have test outputs for the combo of stdout/stderr, for example, named `apply-combined` and `apply-live-combined` and make tests to check those? The rationale is that, by moving some messages from stdout to stderr, we lose some test coverage. I hope we can keep the same degree of test coverage even after this change.

@mumoshu I applied both suggested changes.

**Review by yxxhero — APPROVED** [review]
LGMT

**Comment by yxxhero** [comment]
@mumoshu

**Review by mumoshu — APPROVED** [review]
LGTM. Thank you so much for your amazing work!

**Pull request merged**